doc/dev/doc/extensions/caching.md

```diff
@@ -51,4 +51,20 @@ With these declarations, the Hazelcast cache implementation would be injected in
 
 ## Cache Functionality
 
-Hazelcast's implementation of `IDistributedCache` supports all functionality of the interface. Note however that, due to the inherently asynchronous nature of the Hazelcast client, all synchronous cache operations have to be implemented over asynchronous operations using the `task.GetAwaiter().GetResult()` pattern. This pattern can cause potential threading problems, and should be avoided as much as possible. Always use the asynchronous API wherever possible.
\ No newline at end of file
+Hazelcast's implementation of `IDistributedCache` supports all functionality of the interface. Note however that, due to the inherently asynchronous nature of the Hazelcast client, all synchronous cache operations have to be implemented over asynchronous operations using the `task.GetAwaiter().GetResult()` pattern. This pattern can cause potential threading problems, and should be avoided as much as possible. Always use the asynchronous API wherever possible.
+
+## ASP.NET Core Session State Provider
+`IDistributedCache` implementation can be used to store session state in Hazelcast. After registering 
+Hazelcast `IDistributedCache` implementation to DI (as described above) you can enabled session state management 
+in your ASP.NET Core project. 
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddHazelcastCache();
+var app = builder.Build();
+app.UseSession(); // Session state will be stored in distributed cache provided by AddHazelcastCache()
+```
+> [!WARNING]
+> While reading `ISession` object, it is strongly advised to use async extensions which are provided with this package under `Hazelcast.Caching.Session` namespace. Otherwise, you may not benefit 
```

Comment:
Maybe: When the `ISession` object is supported by the Hazelcast client, its synchronous operations should be avoided, as the Hazelcast client is fully asynchronous and therefore calls to `task.GetAwaiter().GetResult()` may be involved and, potentially, cause threading issues. Prefer the asynchronous operations, such as those provided in the `SessionExtensions` class.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/commit/535f7e6a3a83daba26e75bf2a0903c63fbf3ed73.